lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,179 @@
+      display_name_id = create_display_name
+
+      new_template = parse_result(@conn.post('/api/v2/RelationshipTemplates/Own') do |req|
+        req.body = relationship_template(display_name_id, nbp_uid).to_json
+      end)
+
+      Rails.logger.debug { "Enmeshed::ConnectorApi RelationshipTemplate created: #{new_template[:truncatedReference]}" }
+      new_template[:truncatedReference]
+    end
+
+    def create_display_name
+      new_attribute = parse_result(@conn.post('/api/v2/Attributes') do |req|
+        req.body = {content: display_name_attribute}.to_json
+      end)
+
+      new_attribute[:id]
+    end
+
+    def pending_relationship_for_nbp_uid(nbp_uid)
+      relationships = parse_result(@conn.get('/api/v2/Relationships') do |req|
+        req.params['status'] = 'Pending'
+      end)
+
+      relationships = relationships.select do |rel|
+        if DateTime.parse(rel[:template][:expiresAt]) < 1.day.ago
+          reject_relationship(rel)
+          false
+        else
+          true
+        end
+      end
+
+      relationships.find {|rel| rel.dig(:template, :content, :metadata, :nbp_uid) == nbp_uid }
+    end
+
+    def accept_relationship_change(relationship_id, change_id)
+      respond_to_rel_change(relationship_id, change_id, 'Accept')
+    end
+
+    def reject_relationship_change(relationship_id, change_id)
+      respond_to_rel_change(relationship_id, change_id, 'Reject')
+    end
+
+    def respond_to_rel_change(relationship_id, change_id, action = 'Accept')
+      response = @conn.put("/api/v2/Relationships/#{relationship_id}/Changes/#{change_id}/#{action}") do |req|
+        req.body = {content: {}}.to_json
+      end
+      Rails.logger.debug do
+        "Enmeshed::ConnectorApi responded to RelationshipChange with: #{action}; connector response status is #{response.status}"
+      end
+
+      response.status == 200
+    end
+
+    def reject_relationship(relationship)
+      Rails.logger.debug { "Enmeshed::ConnectorApi rejecting Relationship for template #{relationship[:template][:truncatedReference]}" }
+
+      relationship[:changes].each do |change|
+        reject_relationship_change(relationship[:id], change[:id])
+      end
+    end
+
+    def parse_userdata_in_rel_change(relship_change_json) # rubocop:disable Metrics/AbcSize
+      change_response_items = relship_change_json[:request][:content][:response][:items]
+
+      read_attribute_accept_response_items = change_response_items.select {|item| item[:@type] == 'ReadAttributeAcceptResponseItem' }
+
+      @enmeshed_user_attributes = {}
+
+      read_attribute_accept_response_items.each do |item|
+        attr_type = item[:attribute][:value][:@type]
+        attr_value = item[:attribute][:value][:value]
+
+        @enmeshed_user_attributes[attr_type] = attr_value
+      end
+      user_attributes
+    rescue NoMethodError
+      raise ConnectorError.new("Could not parse userdata in relationship change: #{relship_change_json}")
+    end
+
```

Comment:
At least starting from here, most methods following could (and should) be private, I'd say.

Reply:
Class methods can be private in Ruby, too. [Here](https://stackoverflow.com/a/4953043) is a short answer on how to do that.